### PR TITLE
[feat] Add CIDR transformation for single IP addresses in firewall rules

### PR DESCRIPTION
**Pull request by komer3**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: controller/linodefirewall_controller_helpers.go [hunk 1]
```diff
@@ -157,11 +180,17 @@ func processACL(firewall *infrav1alpha2.LinodeFirewall) (
 		ruleIPv6s := []string{}
 
 		if rule.Addresses.IPv4 != nil {
-			ruleIPv4s = append(ruleIPv4s, *rule.Addresses.IPv4...)
+			// Transform each IPv4 address
+			for _, ip := range *rule.Addresses.IPv4 {
+				ruleIPv4s = append(ruleIPv4s, transformToCIDR(ip))
```
Comment (rahulait) [inline controller/linodefirewall_controller_helpers.go]: Overall the code change looks good to me, not sure if we should validate if its valid ipv4 here or its getting done later on.

Comment (komer3) [reply controller/linodefirewall_controller_helpers.go]: I was thinking of letting linode api do it? What are your thoughts?

Comment (rahulait) [reply controller/linodefirewall_controller_helpers.go]: Sounds good to me. I got confused by the comment `// If not a valid IP, return as-is (will be validated later)` and thought if we missed validating.

**Review by rahulait — APPROVED** [review]
lgtm

**Pull request merged**